client/go/outline/event/event.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package event provides a thread-safe mechanism for managing and triggering events.
+package event
+
+import (
+	"log/slog"
+	"slices"
+	"sync"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/callback"
+)
+
+// EventName is a type alias for string that represents the name of an event.
+// Using a dedicated type improves type safety when working with event names.
+type EventName string
+
+var (
+	mu        sync.RWMutex
+	listeners = make(map[EventName][]callback.Token)
```

Comment:
Again, please avoid globals. Create an object to hold the fields and methods to manage the listeners.